src/scalems/radical/raptor.py

```diff
@@ -226,33 +188,42 @@ def __init__(self, configuration: Configuration):
                 )
         super(ScaleMSMaster, self).__init__()
 
-    def result_cb(self, requests: typing.Sequence[Request]):
+    def result_cb(self, requests: typing.Sequence[rp.TaskDescription]):
+        """SCALE-MS specific handling of tasks completed by the collaborating Worker(s).
+
+        .. todo:: Clarify TaskDescription fields that are unique to raptor tasks, and the details of the
+            result tuple.
+        """
         for r in requests:
             logger.info('result_cb %s: %s [%s]' % (r.uid, r.state, r.result))
 
-    def request_cb(self, requests: RequestInputList) -> RequestInputList:
+    def request_cb(
+            self,
+            requests: typing.Sequence[rp.TaskDescription]
```

Comment:
The contents of `Master._request_cb()` imply that *requests* is a list of Tasks. It looks for a `'uid'` field and passes the list to `Master.submit_tasks()`, which expects either a list of TaskDescription instances or "task dictionaries", which is not clearly defined. However, in `Master._submit_executable_tasks()`, a task description is extracted from the `'description'` field (as noted above), but this is _only_ used for the `'uid'`, whereas the rest of the object us treated as a Task with dictionary-like behavior.

@andre-merzky can you please clarify?

The reason this should be more clearly documented is so that an extension author (like me) knows which objects (or fields) are expected to be convertible (or can be assumed to be valid input) to specific types, which fields to expect to find in an object, and what sort of pre- or post- validation can/should be done.

Currently, I'm not sure
* if there is or should be a difference between `request['uid']` and `request['description']['uid']`
* whether `request['description']` is part of the specified API, such that I should always expect it to map to dictionary representations of the original `TaskDescription` (from the client), or whether parts of *request* are more strongly or weakly specified and subject to change.

Reply:
- you never need to specify `description['uid']` - we only do this in the examples to make them more readable.  Either way though, the `task_dict['uid']` will always be set and is the normative UID of the task.
  - `task_dict['description']` should always be present and is a dict representation of the original task description.

Reply:
`request_cb` receives a sequence of dictionaries that contain a `description` field with the original TaskDescription. The *args* and *kwargs* are fields in the original TaskDescription.

Reply:
Note that the original TaskDescription may not have a `uid`, but the task dictionary will have the canonical uid.